test/extended/images/dryrun.go

```diff
@@ -76,4 +78,26 @@ var _ = g.Describe("[sig-imageregistry] Image --dry-run", func() {
 		o.Expect(err).NotTo(o.HaveOccurred())
 		o.Expect(output).To(o.BeEmpty())
 	})
+
+	g.It("should not create resources [apigroup:image.openshift.io]", func() {
+		g.By("triggering create operation of imagestream with --dry-run=server")
+		err := oc.Run("create").Args("imagestream", "dryrun-test", "--dry-run=server").Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// wait for 30s to make sure we'll not miss any resource created accidentally
+		err = wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 30*time.Second, true, func(ctx context.Context) (bool, error) {
```

Comment:
This guarantees that test will return after 30 seconds. Maybe it is better to return just after 10 seconds.